src/docfx/gc/GarbageCollector.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class GarbageCollector
+    {
+        private const int MaxKeepingDays = 15;
+
+        public static async Task Collect(int notAccessedForDays)
+        {
+            notAccessedForDays = notAccessedForDays <= 0 ? MaxKeepingDays : notAccessedForDays;
+            await CollectGit(notAccessedForDays);
+            CollectUrls(notAccessedForDays);
+        }
+
+        private static async Task CollectGit(int notAccessedForDays)
+        {
+            if (!Directory.Exists(AppData.GitRestoreDir))
+            {
+                return;
+            }
+
+            var gitWorkTreeRoots = Directory.EnumerateDirectories(AppData.GitRestoreDir, ".git", SearchOption.AllDirectories);
+
+            using (Progress.Start("GC dependency git repositories"))
+            {
+                await ParallelUtility.ForEach(
+                    gitWorkTreeRoots,
+                    async gitWorkTreeRoot =>
```

Comment:
can this be extracted to a function ?

Reply:
sure

---
In reply to: [230655577](https://github.com/dotnet/docfx/pull/3571#discussion_r230655577) [](ancestors = 230655577)